packages/neon-api/src/provider/notifications/class.ts

```diff
@@ -0,0 +1,48 @@
+import { logging, settings } from "@cityofzion/neon-core";
+import WebSocket from 'isomorphic-ws';
+import { NotificationMessage } from "./responses";
+const log = logging.default("api");
+
+export class Notifications {
+  private url: string;
+
+  public get name(): string {
+    return `Notifications[${this.url}]`;
+  }
+
+  private websocketSubscriptions: Map<string, WebSocket[]>;
+
+  public constructor(url: string) {
+    if (settings.networks[url] && settings.networks[url].extra.notifications) {
+      this.url = settings.networks[url].extra.notifications;
+    } else {
+      this.url = url;
+    }
+	this.websocketSubscriptions = new Map<string, WebSocket[]>();
+    log.info(`Created Notifications Provider: ${this.url}`);
+  }
+
+  public subscribe(contract: string, callback: (message: NotificationMessage) => void) {

```

Comment:
Personally, when designing APIs I believe the most important thing is to adhere to standards that developers are already familiar with, as that makes the API much more easier to use in my experience. With that in mind, I came up with three different models when designing this api:
1. The model on this PR: `subscribe(contract, callback), unsubscribe(contract)`
2. The standard subscription model used in DOM APIs: `subscribe(contract, callback), unsubscribe(contract, callback)`
3. The model you proposed: `a = subscribe(contract, callback); a.unsubscribe();`

While the 3rd model is the one that I find the most elegant, I believe that the other two would be more intuitive to the developers using neon-js, because:
- 2nd model: A lot of them are probably familiar with the DOM api, given that they are using a js library
- 1st model: This is the model used by [the dAPI](https://neologin.io/api/#event-methods). I didn't design it and I believe it's not the best model but in the future I expect that a lot of the devs using neon-js on the front-end will also have to deal with the dAPI, so there's some value in having consistency there.
- 1st model: I searched "pubsub javascript" on google and two of the first results that came up were tutorials that taught how to implement a pubsub system based on the 1st model

Because of this, I chose to go with the 1st model, but I have no problem changing it if it makes more sense.

Reply:
> Subscribe to all contract events using null

The reason why I didn't account for this use-case is because I believe almost nobody will use it like this, but the solution you proposed (using null) doesn't pollute the api so I'm down for adding that.